include/uapi/sound/sof/tokens.h

```diff
@@ -54,6 +54,7 @@
 #define SOF_TKN_SCHED_DYNAMIC_PIPELINE		206
 #define SOF_TKN_SCHED_LP_MODE			207
 #define SOF_TKN_SCHED_MEM_USAGE			208
+#define SOF_TKN_SCHED_USE_CHAIN_DMA		209
```

Comment:
The commit message needs to be rewritten, too verbose and unclear at the same time

suggested edits:

Add logic for setting up and tearing down chained DMA connections. 

Since pipelines are not used, all the logic to set the pipeline states can be bypassed, with only the DMA programming sequences remaining. In addition the same format needs to be used for host- and link-DMA, without the usual fixup to use the S32_LE format on the link.

Note however that for convenience and compatibility with existing definitions, the topology relies on the concept of pipelines with a 'USE_CHAIN_DMA' token indicating that all the logic shall be bypassed.

Unlike 'normal' ALSA sequences, the chain DMA is not programmed in hw_params/hw_free. The IPC message tp set-up and tear-down chained DMA are sent in sof_ipc4_trigger_pipelines(), but the contents prepared earlier.

Chained DMA is only supported by the Intel HDA DAI for now.

Reply:
Thanks! Certainly better than my version that has evolved over time, and where I've avoided subjects that I am not certain of. With your permission, I take that into use.